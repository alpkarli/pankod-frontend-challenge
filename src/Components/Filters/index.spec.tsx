import { Filters } from "@Components";
import {
	act,
	create,
	ReactTestInstance,
	ReactTestRenderer,
} from "react-test-renderer";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
	useDispatch: () => mockDispatch,
}));

describe("Filters", () => {
	let component: ReactTestRenderer;
	let instance: ReactTestInstance;
	beforeEach(() => {
		act(() => {
			component = create(<Filters />);
		});
		instance = component.root;
	});
	it("should match snapshot", () => {
		expect(component.toJSON()).toMatchSnapshot();
	});
	it("should handle select titleDesc", () => {
		const event = {
			preventDefault() {},
			target: { value: "titleDesc" },
		};
		const select = instance.findByType("select");
		act(() => {
			select.props.onChange(event);
		});
		expect(select.props.value).toBe("titleDesc");
	});
	it("should handle select yearDesc", () => {
		const event = {
			preventDefault() {},
			target: { value: "yearDesc" },
		};
		const select = instance.findByType("select");
		act(() => {
			select.props.onChange(event);
		});
		expect(select.props.value).toBe("yearDesc");
	});
	it("should handle select yearAsc", () => {
		const event = {
			preventDefault() {},
			target: { value: "yearAsc" },
		};
		const select = instance.findByType("select");
		act(() => {
			select.props.onChange(event);
		});
		expect(select.props.value).toBe("yearAsc");
	});
	it("should handle select titleAsc", () => {
		const event = {
			preventDefault() {},
			target: { value: "titleAsc" },
		};
		const select = instance.findByType("select");
		act(() => {
			select.props.onChange(event);
		});
		expect(select.props.value).toBe("titleAsc");
	});
	it("should handle text change", () => {
		const event = {
			preventDefault() {},
			target: { value: "search" },
		};
		const input = instance.findByType("input");
		act(() => {
			input.props.onChange(event);
		});
		expect(input.props.value).toBe("search");
	});
	/* it("should handle text change if length lesser than 3", () => {
		const value = document.getElementById('value')
		const event = {
			preventDefault() {},
			target: { value: "se" },
		};
		const input = instance.findByType("input");
		act(() => {
			input.props.onChange(event);
		});
		
		expect(input.props.value).toBe("se");
	}); */
});
