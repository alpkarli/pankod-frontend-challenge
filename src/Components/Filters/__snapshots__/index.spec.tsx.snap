// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should match snapshot 1`] = `
<div
  className="filters"
>
  <div
    className="content-container"
  >
    <div
      className="filters__content"
    >
      <div
        className="col-xl-4 col-lg-8 col-md-12"
      >
        <div
          className="input-group"
        >
          <input
            aria-describedby="Search"
            aria-label="Search"
            className="form-control"
            onChange={[Function]}
            placeholder="Search..."
            type="text"
            value=""
          />
          <div
            className="input-group-append"
          >
            <button
              className="btn btn-primary"
              type="button"
            >
              <img
                className="lens"
                src="images/lens.svg"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        className="col-xl-3 col-lg-4 col-md-12 mt-4 mt-lg-0"
      >
        <select
          className="form-control"
          id="filters"
          onChange={[Function]}
          value="yearAsc"
        >
          <option
            value="yearAsc"
          >
            Sort by year in asc
          </option>
          <option
            value="yearDesc"
          >
            Sort by year in desc
          </option>
          <option
            value="titleAsc"
          >
            Sort by title in asc
          </option>
          <option
            value="titleDesc"
          >
            Sort by title in desc
          </option>
        </select>
      </div>
    </div>
  </div>
</div>
`;
